pkg/profiler/cpu/bpf/maps/maps.go

```diff
@@ -1485,6 +1506,36 @@ func (m *Maps) RefreshProcessInfo(pid int, shouldUseFPByDefault bool) {
 	}
 }
 
+func (m *Maps) GetUnwindFailedReasons() (map[int]profiler.UnwindFailedReasons, error) {
+	ret := make(map[int]profiler.UnwindFailedReasons)
+	it := m.unwindFailedReasons.Iterator()
+	for it.Next() {
+		key := it.Key()
+		var pid int32
+
+		if err := binary.Read(bytes.NewBuffer(key), m.byteOrder, &pid); err != nil {
+			return nil, err
+		}
+		val, err := m.unwindFailedReasons.GetValue(unsafe.Pointer(&key[0]))
+		if err != nil {
+			return nil, err
+		}
+
+		buf := bytes.NewBuffer(val)
+		reasons := new(profiler.UnwindFailedReasons)
+		v := reflect.ValueOf(reasons)
+		for i := 0; i < v.Elem().NumField(); i++ {
+			fv := v.Elem().Field(i)
+			if err := binary.Read(buf, m.byteOrder, fv.Addr().Interface()); err != nil {
```

Comment:
You can just call a single `binary.Read` on the whole struct, no need for reflection here.